app/src/test/java/com/fstyle/structure_android/data/source/UserRepositoryTest.java

```diff
@@ -29,78 +28,62 @@
     private static final String USER_LOGIN_1 = "abc";
     private static final String USER_LOGIN_2 = "def";
 
-    @Mock
-    NameApi mNameApi;
-
+    @InjectMocks
     private UserRepository mUserRepository;
+    @Mock
+    UserLocalDataSource mLocalDataSource;
+    @Mock
+    UserRemoteDataSource mRemoteDataSource;
 
     @Before
     public void setUp() throws Exception {
         MockitoAnnotations.initMocks(this);
-        mUserRepository = new UserRepository(null, new UserRemoteDataSource(mNameApi));
-    }
-
-    @After
-    public void tearDown() throws Exception {
-
     }
 
     @Test
-    public void searchUsers200ResponseCodeInvokesCorrectApiCalls() {
+    public void searchUsers_200ResponseCode_InvokesCorrectApiCalls() {
+        User user = new User(USER_LOGIN_1);
+        User user2 = new User(USER_LOGIN_2);
+
+        List<User> userReturns = new ArrayList<>();
+        userReturns.add(user);
+        userReturns.add(user2);
         // Given
-        Mockito.when(
-                mNameApi.searchGithubUsers(ArgumentMatchers.anyInt(), ArgumentMatchers.anyString()))
-                .thenReturn(Observable.just(githubUserList()));
+        Mockito.when(mRemoteDataSource.searchUsers(ArgumentMatchers.anyString(),
+                ArgumentMatchers.anyInt())).thenReturn(Observable.just(userReturns));
 
         // When
-        TestSubscriber<UsersList> subscriber = new TestSubscriber<>();
-        mUserRepository.getRemoteDataSource().searchUsers(USER_LOGIN_1, 2).subscribe(subscriber);
+        TestSubscriber<List<User>> subscriber = new TestSubscriber<>();
+        mUserRepository.searchUsers(USER_LOGIN_1, 2).subscribe(subscriber);
 
         // Then
         subscriber.awaitTerminalEvent();
         subscriber.assertNoErrors();
 
-        List<UsersList> onNextEvents = subscriber.getOnNextEvents();
-        UsersList usersList = onNextEvents.get(0);
-        Assert.assertEquals(USER_LOGIN_1, usersList.getItems().get(0).getLogin());
-        Assert.assertEquals(USER_LOGIN_2, usersList.getItems().get(1).getLogin());
-        Mockito.verify(mNameApi).searchGithubUsers(2, USER_LOGIN_1);
-    }
-
-    private UsersList githubUserList() {
-        User user = new User(USER_LOGIN_1);
-
-        User user2 = new User(USER_LOGIN_2);
-
-        List<User> githubUsers = new ArrayList<>();
-        githubUsers.add(user);
-        githubUsers.add(user2);
-        UsersList usersList = new UsersList(githubUsers);
-        return usersList;
+        List<List<User>> onNextEvents = subscriber.getOnNextEvents();
+        List<User> users = onNextEvents.get(0);
+        Assert.assertEquals(USER_LOGIN_1, users.get(0).getLogin());
+        Assert.assertEquals(USER_LOGIN_2, users.get(1).getLogin());
+        Mockito.verify(mRemoteDataSource).searchUsers(USER_LOGIN_1, 2);
     }
 
     @Test
-    public void searchUsersOtherHttpErrorSearchTerminatedWithError() {
+    public void searchUsers_OtherHttpError_SearchTerminatedWithError() {
```

Comment:
ERROR: [MethodNamingConventions] Method names should not contain underscores
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/naming.html#MethodNamingConventions